pyccel/parser/semantic.py

```diff
@@ -2803,6 +2875,16 @@ def _visit_Lambda(self, expr):
                 expr = Lambda(tuple(expr.variables), expr_new)
         return expr
 
+    def _visit_IndexedFunctionCall(self, expr, **settings):
+        name     = expr.funcdef
+        func     = self.scope.find(name, 'functions')
```

Comment:
If you don't call `get_expected_name` then I don't think you will find the semantic object if the function has been renamed

Reply:
is there is any test for that ??

Reply:
fixed 71091e1

Reply:
Have you written one? There are no tests for `IndexedFunctionCall` unless you have written them

Reply:
I agree that it could be useful to add one

Reply:
Please add a test with a name conflict

Reply:
Any progress on this?

Reply:
Sorry, so basically any test will fail, of course. I still need a handle for that. Waiting for your PR to get merged so I can see what is the best way to handle that

Reply:
I'm using get_expected_name as will befoul trying to find the fun node

Reply:
The PR is now merged and the repo has been updated